aspnetcore/blazor/hybrid/class-libraries.md

```diff
@@ -0,0 +1,514 @@
+  <link href="_content/SharedLibrary/css/app.css" rel="stylesheet" />
+  ```
+
+  > [!NOTE]
+  > In a Blazor WebAssembly app, the preceding `<link>` tags are in the `wwwroot/index.html` file.
+
+* Place additional shared components and C# code into the RCL.
+
+  > [!IMPORTANT]
+  > A service can only be moved to the RCL and registered without further changes *if the service doesn't rely on hosting model-specific API*. If a service must be implemented independently by hosting model, see the guidance in the [Provide code and services independent of hosting model](#provide-code-and-services-independent-of-hosting-model) section.
+
+  Apps created from the Blazor Server and .NET MAUI Blazor project templates include a `Pages/FetchData` component that obtains weather data from a registered service. Components can be placed into the RCL's `Pages` folder and shared between the Blazor Server and Blazor Hybrid projects, along with associated C# code, such as the `WeatherForecast` class (`Data/WeatherForecast.cs`) and the `WeatherForecastService` class (`Data/WeatherForecastService.cs`).
+  
+  Update the namespaces to match the RCL (for example, `namespace SharedLibrary.Data`). The shared `FetchData` component in the RCL is updated to use the namespace of the RCL's `Data` folder to access the `WeatherForecastService`. The following example assumes that the `FetchData` component is moved from the Blazor Server app to the RCL with a Blazor Server project name of `BlazorWeb`:
+
+  ```diff
+  - @using BlazorWeb.Data
+  + @using SharedLibrary.Data
+  ```
+  
+  You can move the entire `Data` folder to move the class files if there are no classes in the `Data` folder specific to the Blazor Server project. Each of the projects consuming the RCL in the solution can drop their `Data` folder assets and the components moved to the RCL.
+
+  The Blazor Server and the Blazor Hybrid service registrations register the same service when the RCL provides a common service. In both .NET MAUI Blazor's `MauiProgram.cs` and Blazor Server's `Program.cs`, add the namespace for the RCL's `Data` folder assets and register the RCL's `WeatherForecastService`:
+
+  ```csharp
+  using SharedLibrary.Data;
+
+  ...
+
+  builder.Services.AddSingleton<WeatherForecastService>();
+  ```
+
+  > [!NOTE]
+  > Windows Forms Blazor projects register services in `Form1`'s contstructor (`Form1.cs`). WPF Blazor projects register services in `MainWindow`'s constructor (`MainWindow.xaml`).
+
+  Remove the `Data` folder namespace in the Blazor Hybrid and Blazor Server projects:
+
+  ```diff
+  - using BlazorNative.Data;
+  ```
+
+  ```diff
+  - using BlazorWeb.Data;
+  ```
+
+  For additinal information on how to share static assets across projects, see the following articles:
+
+  * <xref:blazor/components/class-libraries>
+  * <xref:razor-pages/ui-class>
+
+## Provide code and services independent of hosting model
```

Comment:
Services let you abstract away details of the Blazor hosting model and also platform specific details. The `ICustomService` example does a good job of showing how to deal with platform specific functionality (similar to the `INativeAudioService` usage in the .NET Podcasts app). I think it would also be good to show how to deal with abstracting away Blazor hosting model specific concerns. For example, you could show abstracting different weather forecast service implementations, one that makes an HTTP request for Blazor WebAssembly and Blazor Hybrid, and one that accesses the data "directly" for Blazor Server.